src/test/java/org/smartregister/converters/EventConverterTest.java

```diff
@@ -32,8 +45,91 @@ public void testConvertEventToQuestionnaireResponse() {
+
+	@Test
+	public void testConvertEventToQuestionnaireResponseWithMultipleValuesOfObs() {
+		Event event;
+		event = gson.fromJson(EVENT_JSON_1, Event.class);
+		event.setChildLocationId("child-location-id");
+		event.setEventType("event-type");
+		event.setEntityType("entity-type");
+		event.setLocationId("location-id");
+		event.setProviderId("provider-id");
+		event.setBaseEntityId("base-entity-id");
+		event.setTeam("team");
+		event.setTeamId("team-id");
+		QuestionnaireResponse questionnaireResponse = EventConverter.convertEventToEncounterResource(event);
+		assertNotNull(questionnaireResponse);
+		assertEquals(questionnaireResponse.getSubject().getReference().getValue(), event.getBaseEntityId());
+		assertEquals(questionnaireResponse.getStatus().getValueAsEnumConstant().value(), "completed");
+		assertEquals(questionnaireResponse.getAuthor().getReference().getValue(), event.getProviderId());
+		assertEquals(questionnaireResponse.getMeta().getVersionId().getValue(), String.valueOf(event.getServerVersion()));
+		assertEquals(questionnaireResponse.getItem().get(0).getLinkId().getValue(), "locationId");
+		assertEquals(
+				questionnaireResponse.getItem().get(0).getAnswer().get(0).getValue().as(com.ibm.fhir.model.type.String.class)
+						.getValue(), event.getLocationId());
+		assertEquals(questionnaireResponse.getItem().get(1).getLinkId().getValue(), "childLocationId");
+		assertEquals(
+				questionnaireResponse.getItem().get(1).getAnswer().get(0).getValue().as(com.ibm.fhir.model.type.String.class)
+						.getValue(), event.getChildLocationId());
+		assertEquals(questionnaireResponse.getItem().get(2).getLinkId().getValue(), "teamId");
+		assertEquals(
+				questionnaireResponse.getItem().get(2).getAnswer().get(0).getValue().as(com.ibm.fhir.model.type.String.class)
+						.getValue(), event.getTeamId());
+		assertEquals(questionnaireResponse.getItem().get(3).getLinkId().getValue(), "team");
+		assertEquals(
+				questionnaireResponse.getItem().get(3).getAnswer().get(0).getValue().as(com.ibm.fhir.model.type.String.class)
+						.getValue(), event.getTeam());
+		
+		PathEvaluatorLibrary.init(null, null, null);
+		//TODO : Do we need to test individual obs, details and identifier fields?
+		FHIRPathElementNode node = PathEvaluatorLibrary.getInstance().evaluateElementExpression(questionnaireResponse, "QuestionnaireResponse.item.where(linkId='totPopulation')");
+		QuestionnaireResponse.Item totPopulation = node.element().as(QuestionnaireResponse.Item.class);
+		assertEquals(1,totPopulation.getAnswer().size());
+		assertEquals(com.ibm.fhir.model.type.String.of("2"),totPopulation.getAnswer().get(0).getValue());
+		node = PathEvaluatorLibrary.getInstance().evaluateElementExpression(questionnaireResponse, "QuestionnaireResponse.item.where(linkId='existingLLINs')");
+		
+		QuestionnaireResponse.Item existingLLINs = node.element().as(QuestionnaireResponse.Item.class);
+		assertEquals(2, existingLLINs.getAnswer().size());  //Test for multiple values
```

Comment:
I have changed this instead of looping for all the items to use FHIRPath expressions to look up obs

Reply:
Thanks.